server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v67/MigratePreviousAnalysisLeakPeriod.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.platform.db.migration.version.v67;
+
+import java.sql.SQLException;
+import org.sonar.db.Database;
+import org.sonar.db.dialect.Oracle;
+import org.sonar.server.platform.db.migration.step.DataChange;
+import org.sonar.server.platform.db.migration.step.MassUpdate;
+
+public class MigratePreviousAnalysisLeakPeriod extends DataChange {
+
+  private static final String ORACLE_SQL = "select id from properties " +
+    " where prop_key = 'sonar.leak.period' and (text_value='previous_analysis' or dbms_lob.compare(clob_value, to_clob('previous_analysis')) = 0)";
```

Comment:
I really don't like to have specific query => As the number of data related to leak in properties table is not too big, you can remove `and (text_value='previous_analysis'` and filter on "previous_analysis" value in the massUpdate.execute

Reply:
Forget what I've said before => clob_value only contains value greater than 4000 characters => "previous_analysis" is less than 400 characters so you can remove filter on clob_value

Reply:
Yes I know, I wondered if I should or not read the value because functionally if the value is in clob we do not migrate it. I'm fine removing this.

Reply:
I understand, but if my understanding is correct, it could have never happen => Good thing to remove it !